### PR TITLE
Implement support for configservice recorders and rules

### PR DESCRIPTION
**Pull request by zackproser**
## Description

Fixes #000.

## Demo

aws-config

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [x] Update the docs.
- [x] Run the relevant tests successfully, including pre-commit checks.
- [x] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [x] Include release notes. If this PR is backward incompatible, include a migration guide.
- [x] Attention Grunts - if this PR adds support for a new resource, ensure the `nuke_sandbox` and `nuke_phxdevops` jobs in `.circleci/config.yml` have been updated with appropriate exclusions (either directly in the job or via the `.circleci/nuke_config.yml` file) to prevent nuking IAM roles, groups, resources, etc that are important for the test accounts.

## Release Notes (draft)

Added support for inspecting and nuking configservice recorders and rules

### CODE REVIEW

File: aws/config_service_test.go [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+	require.NoError(t, lookupErr)
+
+	// If we have no recorders in the region, create one - otherwise we have no more work to do
+	if len(output.ConfigurationRecorders) == 0 {
+		configRecorderName := createConfigurationRecorder(t, region)
+		require.NotEqual(t, "", configRecorderName)
+		return configRecorderName, nil
+	}
+
+	return aws.StringValue(output.ConfigurationRecorders[0].Name), nil
```
Comment (denis256) [inline aws/config_service_test.go:106]: Noticed that `ensureConfigurationRecorderExistsInRegion()` in all cases return `nil` as error

File: aws/config_service.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+			Identifier:   configRuleName,
+			ResourceType: "Config service rule",
+			Error:        err,
+		}
+		report.Record(e)
+
+		if err != nil {
+			logging.Logger.Debugf("[Failed] %s", err)
+		} else {
+			deletedConfigRuleNames = append(deletedConfigRuleNames, aws.String(configRuleName))
```
Comment (denis256) [inline aws/config_service.go:81]: Value assigned to `deletedConfigRuleNames` is never used, maybe it should be printed after `configRuleNames` processing?

File: aws/config_recorder.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	}
+
+	return config.ShouldInclude(
+		aws.StringValue(configRecorder.Name),
+		configObj.ConfigServiceRecorder.IncludeRule.NamesRegExp,
+		configObj.ConfigServiceRecorder.ExcludeRule.NamesRegExp,
+	)
+}
+
+func nukeAllConfigRecorders(session *session.Session, configRecorderNames []string) error {
```
Comment (denis256) [inline aws/config_recorder.go:48]: Function return `nil` as error each time, make sense to have error in the signature?

File: aws/config_service_test.go (hunk 1 above)
Comment (zackproser) [reply aws/config_service_test.go]: Good catch - fixed in [a488a7d](https://github.com/gruntwork-io/cloud-nuke/pull/405/commits/a488a7dba2e73ec2706fb3bf60b4dc4f9b096efa)

File: aws/config_service.go (hunk 2 above)
Comment (zackproser) [reply aws/config_service.go:78]: Thanks, fixed in 183be69

File: aws/config_recorder.go (hunk 3 above)
Comment (zackproser) [reply aws/config_recorder.go:48]: Probably not, but in order to satisfy the interface for `AwsResource` it needs to :/ I'll think about how to clean this up.

**Review by denis256 — APPROVED** [review]

**Comment by zackproser** [comment]
Thanks for review!

**Pull request merged**